### PR TITLE
Associate user with drop

### PR DESCRIPTION
**Pull request by berlintam**
https://github.com/sounddrop/sounddrop/issues/55
- [x] todo: update spec

### CODE REVIEW

**Comment by berlintam** [comment]
@starkcoffee this is finally working w/ specs passing! If you could give it a look over it would be much appreciated. 

~~There's one issue I'm still struggling with: when you go to edit your drop, the drop's fields aren't pre-filled like they should be. I'm using `render` and everything...~~

~~plus I'd like to pull the place/title section of the drop form out into its own partial & only render that on the edit page, because I guess ppl shouldn't be able to completely change the url/track id of their drop because well... then it's just a different drop. wdyt?~~

Those things above are fixed/done!

File: app/controllers/drops_controller.rb [hunk 1]
```diff
@@ -10,31 +11,55 @@ def new
...
+      redirect_to drop_path(@drop), notice: 'Drop successfully updated'
+    elsif drop_params["sc_track"].present?
+      @drop.update_attributes(drop_params.merge({place_id: @place.id}))
+      redirect_to drop_path(@drop), notice: 'Drop successfully updated'
+    else
+      render :edit
+    end
+  end
+
+  def destroy
```
Comment (berlintam) [inline app/controllers/drops_controller.rb:51]: TODO: check that the person destroying the drop is the same person who made it

File: app/controllers/drops_controller.rb [hunk 2]
```diff
@@ -10,31 +11,55 @@ def new
...
       render :login_redirect
     else
-      client = SoundCloud.new(:access_token => session[:access_token_hash]["access_token"])
-      @current_user = client.get('/me')
-      @current_user_tracks = client.get('/me/tracks')
       @drop = Drop.new
     end
   end
 
+  def edit
```
Comment (berlintam) [inline app/controllers/drops_controller.rb]: TODO: check that the person destroying the drop is the same person who made it

File: app/controllers/drops_controller.rb [hunk 3]
```diff
@@ -1,6 +1,10 @@
 require 'soundcloud'
 
 class DropsController < ApplicationController
+  include ApplicationHelper
+
+  before_action :set_drop, only: [:show, :edit, :update, :destroy]
+  before_action :require_creator_of_drop, except: [:index, :show, :new, :create]
```
Comment (starkcoffee) [inline app/controllers/drops_controller.rb:7]: nice use of except!

File: app/controllers/drops_controller.rb [hunk 4]
```diff
@@ -58,6 +56,18 @@ def place_params
       params.require(:drop).require(:place).permit(:name, :longitude, :latitude, :location)
     end
 
+    def client
+      @client ||= SoundCloud.new(:access_token => session[:access_token_hash]["access_token"])
```
Comment (starkcoffee) [inline app/controllers/drops_controller.rb]: nice memoisation!

File: app/models/drop.rb [hunk 5]
```diff
@@ -18,4 +18,12 @@ def soundcloud_track
...
     client.get("/tracks/#{self.sc_track}")
   end
+
+  def place_name
+    place.try(:name)
+  end
+
+  def place_location
+    place.try(:location)
+  end
```
Comment (starkcoffee) [inline app/models/drop.rb:28]: handy!

File: spec/controllers/drops_controller_spec.rb [hunk 6]
```diff
@@ -1,52 +1,153 @@
...
+          expect(drop.place_id).to eql 5
+          expect(drop.place.name).to eql 'A new place'
+          expect(drop.place.location).to eql 'Here, Berlin'
+          expect(drop.sc_track).to eql 345
+          expect(drop.title).to eql 'A new place'
+          expect(response).to redirect_to drop_path(drop)
+        end
+      end
+
+      describe 'it updates the drop when a url is passed in' do
```
Comment (starkcoffee) [inline spec/controllers/drops_controller_spec.rb]: this and the prev test are now the same

**Comment by starkcoffee** [comment]
👍 ⚡ epic!

**Pull request merged**